### PR TITLE
fix(import-populator): Make copying of annotations more robust

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

Similar to 2fa5f27, make copying of annotations from a prime PVC to a target PVC more robust, by moving it before rebinding the PV from prime to target. This way we can ensure the annotations are already present once the PVC becomes ready.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by 0xFelix** [comment]
@akalenyu PTAL, not sure this is 100% required.

**Comment by 0xFelix** [comment]
@akalenyu Do you think this is still needed?

**Comment by akalenyu** [comment]
> @akalenyu Do you think this is still needed?

I think so, yeah
It's completely possible that the annotations only make it after the target is rebound

**Comment by 0xFelix** [comment]
@akalenyu It looks better now, wdyt?

**Comment by mhenriks** [comment]
looks good I think but I'll let @alromeros be the judge

File: pkg/controller/populators/import-populator.go [hunk 1]
```diff
@@ -160,7 +160,10 @@ func (r *ImportPopulatorReconciler) reconcileTargetPVC(pvc, pvcPrime *corev1.Per
 		}
 
 		if cc.IsPVCComplete(pvcPrime) && cc.IsUnbound(pvc) {
-			// Once the import is succeeded, we copy labels and rebind the PV from PVC to target PVC
+			// Once the import is succeeded, we copy annotations and labels and rebind the PV from PVC to target PVC
```
Comment (alromeros) [inline pkg/controller/populators/import-populator.go:163]: We will update the annotations again after the rebind, are we sure we need that? Having three `updatePVCWithPVCPrimeAnnotations` calls in the same function makes this harder to maintain. It's not this PR's fault though. Maybe we could move the two updateAnnotations and updateLabels just above the switch?

Comment (alromeros) [reply pkg/controller/populators/import-populator.go:163]: Nevermind, `UpdatesMultistageImportSucceeded` updates the PVC prime annotations and we would need to call it afterwards

Comment (alromeros) [reply pkg/controller/populators/import-populator.go:163]: What about having one at the top before entering the switch and another one after checking `UpdatesMultistageImportSucceeded`? That would leave us with two calls. Not a must though, I'll lgtm and we can consider improving this code a bit in the near future.

**Comment by alromeros** [comment]
Thanks!
/lgtm

**Pull request merged**